bundles/binding/org.openhab.binding.panstamp/src/main/java/org/openhab/binding/panstamp/internal/PanStampBinding.java

```diff
@@ -0,0 +1,566 @@
+				setup.setChannel(cfg.networkChannel);
+			}
+			if (cfg.networkSecurityOption != -1) {
+				// FIXME
+			}
+			if ((cfg.networkId != -1) || (cfg.networkChannel != -1)
+					|| (cfg.networkDeviceAddress != -1)) {
+				modem.setSetup(setup);
+				logger.info(
+						"Modem setup updated to address {}, channel {}, and network ID {}",
+						setup.getDeviceAddress(), setup.getChannel(),
+						setup.getNetworkID());
+			}
+		} catch (SwapException e) {
+			logger.error("Error configuring network: {}", e.getMessage());
+		}
+	}
+
+	private void setupDirectories(PanStampBindingSettings cfg) {
+		File xmlDir = new File(cfg.xmlDir);
+		if (xmlDir.exists() && xmlDir.isDirectory() && xmlDir.canRead()) {
+			File devFile = new File(xmlDir, "devices.xml");
+			if (devFile.exists() && devFile.canRead() && devFile.isFile()) {
+				network.setDeviceLibrary(new CompoundDeviceLibrary(
+						new FileLibrary(xmlDir), new ClassLoaderLibrary()));
+				logger.info("Configured XML directory to {}", cfg.xmlDir);
+			} else {
+				logger.error(
+						"Devices file in XML directory {} can not be read",
+						cfg.xmlDir);
+			}
+		} else {
+			logger.error("XML directory {} can not be read", cfg.xmlDir);
+		}
+	}
+
+	private void updateProductCode(final PanStampBindingConfig<?> conf) {
+		store.addProductCode(conf.getAddress(), conf.getManufacturerId(),
+				conf.getProductId());
+		logger.debug("Product code for {} updated to {}/{}", conf.getAddress(),
+				conf.getManufacturerId(), conf.getProductId());
+
+	}
+
+	/* Register a detected panStamp device with the binding */
+	private void addDevice(PanStamp dev) {
+		try {
+			logger.debug("addDevice({} ({}/{})", dev.getAddress(),
+					dev.getManufacturerId(), dev.getProductId());
+		} catch (NetworkException e) {
+			logger.error(e.getMessage());
+		}
+		dev.addListener(panStampListener);
+		for (Register reg : dev.getRegisters()) {
+			if (!reg.isStandard())
+				addRegister(reg);
+		}
+	}
+
+	/* Unregiter a panStamp device from the binding */
+	private void removeDevice(PanStamp dev) {
+		try {
+			logger.debug("removeDevice({} ({}/{}))", dev.getAddress(),
+					dev.getManufacturerId(), dev.getProductId());
+		} catch (NetworkException e) {
+			logger.error(e.getMessage());
+		}
+		dev.removeListener(panStampListener);
+		for (Register reg : dev.getRegisters()) {
+			if (!reg.isStandard())
+				removeRegister(reg);
+		}
+	}
+
+	private void addRegister(Register reg) {
+		logger.debug("addRegister({}->{})", reg.getDevice().getAddress(),
+				reg.getId());
+		reg.addListener(registerListener);
+		for (Endpoint<?> ep : reg.getEndpoints()) {
+			addEndpoint(ep);
+		}
+	}
+
+	private void removeRegister(Register reg) {
+		reg.removeListener(registerListener);
+		for (Endpoint<?> ep : reg.getEndpoints()) {
+			removeEndpoint(ep);
+		}
+	}
+
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	private void addEndpoint(Endpoint ep) {
+		logger.debug("addEndpoint({}->{})", PanStampConversions.toString(ep));
```

Comment:
The debug formatting here was fixed, and I put in some effort to make it consistent in other places.